churaverse-plugins-server/src/churarenPlugin/churarenAlchemyItems/trapPlugin/trapPlugin.ts

```diff
@@ -0,0 +1,93 @@
+import { initTrapPluginStore } from './store/initTrapPluginStore'
+import { Trap } from './domain/trap'
+import { removeDieTrap } from './domain/trapService'
+import { RegisterOnOverlapEvent } from '@churaverse/collision-detection-plugin-server/event/registerOnOverlap'
+import { Boss } from '@churaverse/churaren-boss-plugin-server/domain/boss'
+import '@churaverse/churaren-boss-plugin-server/store/defBossPluginStore'
+import { TrapDamageCause } from './domain/trapDamageCause'
+
+export class TrapPlugin extends BaseGamePlugin {
+  public gameId = CHURAREN_CONSTANTS.GAME_ID
+  private trapPluginStore!: TrapPluginStore
+  private networkPluginStore!: NetworkPluginStore<IMainScene>
+  private socketController?: SocketController
+
+  public listenEvent(): void {
+    super.listenEvent()
+    this.bus.subscribeEvent('init', this.init.bind(this))
+
+    this.socketController = new SocketController(this.bus, this.store)
+    this.bus.subscribeEvent('registerMessage', this.socketController.registerMessage.bind(this.socketController))
+    this.bus.subscribeEvent(
+      'registerMessageListener',
+      this.socketController.setupRegisterMessageListener.bind(this.socketController)
+    )
+
+    this.bus.subscribeEvent('registerOnOverlap', this.registerOnOverlapBoss.bind(this))
+  }
+
+  private init(): void {
+    initTrapPluginStore(this.store)
+    this.networkPluginStore = this.store.of('networkPlugin')
+    this.trapPluginStore = this.store.of('trapPlugin')
+  }
+
+  protected subscribeGameEvent(): void {
+    super.subscribeGameEvent()
+    this.bus.subscribeEvent('entitySpawn', this.spawnTrap)
+  }
+
+  protected unsubscribeGameEvent(): void {
+    super.unsubscribeGameEvent()
+    this.bus.unsubscribeEvent('entitySpawn', this.spawnTrap)
+  }
+
+  protected handleGameStart(): void {
+    this.socketController?.registerMessageListener()
+  }
+
+  protected handleGameTermination(): void {
+    this.socketController?.unregisterMessageListener()
+    this.removeAllTrap()
+  }
+
+  private readonly spawnTrap = (ev: EntitySpawnEvent): void => {
+    if (!(ev.entity instanceof Trap)) return
+    const trap = ev.entity
+    this.trapPluginStore.traps.set(trap.trapId, trap)
+  }
+
+  private removeAllTrap(): void {
+    const trapIds = this.trapPluginStore.traps.getAllId()
+    trapIds.forEach((trapId) => {
+      this.trapPluginStore.traps.delete(trapId)
+    })
+  }
+
+  private registerOnOverlapBoss(ev: RegisterOnOverlapEvent): void {
+    ev.collisionDetector.register(
+      this.trapPluginStore.traps,
+      this.store.of('bossPlugin').bosses,
+      this.trapHit.bind(this)
+    )
+  }
+
+  private trapHit(trap: Trap, boss: Boss): void {
+    if (boss.isDead) return
+    trap.die()
+    // トラップ発動イベントの発火
+    const trapDamageCause = new TrapDamageCause(trap)
+    const livingDamageEvent = new LivingDamageEvent(boss, trapDamageCause, trap.power)
+    this.bus.post(livingDamageEvent)
+    removeDieTrap(this.trapPluginStore.traps, trap.trapId)
```

Comment:
こちらの呼び出している関数の処理自体は単純なので、関数で呼び出さずに書いた方が処理がまたがることがないので、読みやすくなると思います。

```suggestion
    this.trapPluginStore.traps.delete(trap.trapId)
```